src/app/click-handler.directive.ts

```diff
@@ -0,0 +1,31 @@
+import { Directive, HostListener, Input } from '@angular/core';
+import { ConfigService } from './config.service';
+
+@Directive({
+  selector: '[appClickHandler]'
+})
+export class ClickHandlerDirective {
+
+  @Input('appClickHandler')
+  private appClickHandler: () => void;
+
+  constructor(private config: ConfigService) { }
+
+  @HostListener('mouseup') onRelease() {
```

Comment:
I think it would be nice to have a public static enum of these event strings that we could use in various places

Reply:
I think it would be too :)
I shall add one shortly